tests/test_main.py

```diff
@@ -0,0 +1,157 @@
+                },
+                {
+                    "id": "ingredient_3",
+                    "abstract_ingredient_id": "abstract_3",
+                    "recipe_id": "fruit_pie_id",
+                    "name": "cane sugar",
+                    "quantity": 200,
+                    "units": "g",
+                },
+                {
+                    "id": "ingredient_4",
+                    "abstract_ingredient_id": "abstract_4",
+                    "recipe_id": "fruit_pie_id",
+                    "name": "goat butter",
+                    "quantity": 1,
+                    "units": "stick",
+                },
+            ],
+            "instructions": "Combine all ingredients in pot",
+            "servings": 1,
+            "source": "http://botw-recipes.com/",
+        }
+
+
+class MockPilafRecipe:
+    @staticmethod
+    def to_json():
+        date = datetime.date(2023, 1, 1)
+        return {
+            "id": "gourmet_poultry_pilaf_id",
+            "created_at": date,
+            "edited_at": date,
+            "name": "gourmet poultry pilaf",
+            "ingredients": [
+                {
+                    "id": "ingredient_5",
+                    "abstract_ingredient_id": "abstract_5",
+                    "recipe_id": "gourmet_poultry_pilaf_id",
+                    "name": "hylian rice",
+                    "quantity": 350,
+                    "units": "g",
+                },
+                {
+                    "id": "ingredient_6",
+                    "abstract_ingredient_id": "abstract_6",
+                    "recipe_id": "gourmet_poultry_pilaf_id",
+                    "name": "bird egg",
+                    "quantity": 2,
+                    "units": "ea",
+                },
+                {
+                    "id": "ingredient_7",
+                    "abstract_ingredient_id": "abstract_4",
+                    "recipe_id": "gourmet_poultry_pilaf_id",
+                    "name": "goat butter",
+                    "quantity": 0.5,
+                    "units": "stick",
+                },
+                {
+                    "id": "ingredient_8",
+                    "abstract_ingredient_id": "abstract_8",
+                    "recipe_id": "gourmet_poultry_pilaf_id",
+                    "name": "raw whole bird",
+                    "quantity": 1,
+                    "units": "ea",
+                },
+            ],
+            "instructions": "Combine all ingredients in pot",
+            "servings": 1,
+            "source": "http://botw-recipes.com/",
+        }
+
+
+class MockQuery:
+    @staticmethod
+    def get(*args, **kwargs):
+        return MockFruitPieRecipe()
+
+    @staticmethod
+    def all(*args, **kwargs):
+        return [MockFruitPieRecipe(), MockPilafRecipe()]
+
+
+class MockRecipeFromModel:
+    query = MockQuery
+
+
+def test_get_single_recipe(client, monkeypatch):
+    with app.app_context():
+        monkeypatch.setattr(Recipe, "query", MockQuery)
```

Comment:
- Tests aren't called in this file. The dev should run all tests and show a passing result with each PR - I'd like to write scripts to get this automated in both local and in github but we're not there yet :)
- `monkeypatch` is replacing the `query` attribute of the `Recipe` object with our `MockQuery` object. This is so that running `Recipe.query.get` and `Recipe.query.all` don't make a call to the database, but instead uses our predefined methods.

Reply:
`app.app_context` binds the application context for Flask and SQLAlchemy and is needed to prevent an error ;)